### PR TITLE
Unit test to ensure versions are as set

### PR DESCRIPTION
**Pull request by chayim**

### CODE REVIEW

**Comment by chayim** [comment]
@dvirdukhan Can you have a look? IMHO this solve the potential issue we discussed - now validating that module versions match the configuration.

File: tests/smoketest/mixins.py [hunk 1]
```diff
@@ -83,6 +83,35 @@ def test_search(self, r):
...
+        modules = {m.get("name").lower(): m.get("ver") for m in modlist}
+        for k, v in modmap.items():
+            yamlversion = versions.get(k)
+
+            # remap
+            parts = yamlversion.split(".")
+            last = parts[2]
+            if len(parts[2]) == 1:
+                last = f"{parts[2]}0"
+            version = f"{parts[0]}0{parts[1]}{last}"
```
Comment (DvirDukhan) [inline tests/smoketest/mixins.py:110]: Major version is always a single digit?

File: tests/smoketest/mixins.py [hunk 2]
```diff
@@ -83,6 +83,35 @@ def test_search(self, r):
...
+        modlist = r.module_list()
+        modules = {m.get("name").lower(): m.get("ver") for m in modlist}
+        for k, v in modmap.items():
+            yamlversion = versions.get(k)
+
+            # remap
+            parts = yamlversion.split(".")
+            last = parts[2]
+            if len(parts[2]) == 1:
+                last = f"{parts[2]}0"
```
Comment (DvirDukhan) [inline tests/smoketest/mixins.py:109]: Why `0` in the head and not in the tail?
how about minor versions?

File: tests/smoketest/mixins.py (hunk 1 above)
Comment (chayim) [reply tests/smoketest/mixins.py:110]: So far... when decoding what we have, yes.

File: tests/smoketest/mixins.py (hunk 2 above)
Comment (chayim) [reply tests/smoketest/mixins.py:109]: This is the modules - not us. I don't know why you versioned them this way :p

**Review by ofekshenawa — APPROVED** [review]

**Pull request merged**